src/main/java/net/fabricmc/loom/build/mixin/AnnotationProcessorInvoker.java

```diff
@@ -141,8 +141,7 @@ public void configureMixin() {
 				project.getLogger().info("Adding mixin to classpath of AP config: " + processorConfig.getName());
 				// Pass named MC classpath to mixin AP classpath
 				processorConfig.extendsFrom(
-						configs.getByName(minecraftSourceSets.getCombinedSourceSetName()),
-						configs.getByName(Constants.Configurations.MOD_COMPILE_CLASSPATH_MAPPED),
+						configs.getByName(Constants.Configurations.LOADER_DEPENDENCIES),
 						configs.getByName(Constants.Configurations.MAPPINGS_FINAL)
```

Comment:
I don't think the `mappingsFinal` configuration is needed here? It seems like the mappings file used is passed below as a normal path, and the Mixin AP graps it from that file rather than a classpath resource.

Admittedly I don't think it'd cause too much harm, but if the AP class loader holds on to the file handle it might prevent changes to locally generated mappings.